demos/python_demos/common/monitors_extension/monitors_extension.cpp

```diff
@@ -0,0 +1,176 @@
+        return nullptr;
+    }
+    try {
+        return PyUnicode_FromFormat(self->_presenter->reportMeans().c_str());
+    } catch (std::exception &exception) {
+        PyErr_SetString(PyExc_RuntimeError, exception.what());
+        return nullptr;
+    }
+}
+
+PyMethodDef presenter_methods[] = {
+        {"handleKey", reinterpret_cast<PyCFunction>(presenter_handleKey), METH_VARARGS | METH_KEYWORDS},
+        {"drawGraphs", reinterpret_cast<PyCFunction>(presenter_drawGraphs), METH_VARARGS | METH_KEYWORDS},
+        {"reportMeans", reinterpret_cast<PyCFunction>(presenter_reportMeans), METH_NOARGS},
+        {}};  // Sentinel
+
+PyObject *presenter_getYPos(PresenterObject *self, void *closure) {
+    if (!self->_presenter) {
+        PyErr_SetString(PyExc_AssertionError, "Underlying C++ presenter is nullptr");
+        return nullptr;
+    }
+    return PyLong_FromLong(self->_presenter->yPos);
+}
+
+PyObject *presenter_getGraphSize(PresenterObject *self, void *closure) {
+        if (!self->_presenter) {
+            PyErr_SetString(PyExc_AssertionError , "Underlying C++ presenter is nullptr");
+            return nullptr;
+        }
+        return Py_BuildValue("ii", self->_presenter->graphSize.width, self->_presenter->graphSize.height);
+}
+
+PyObject *presenter_getGraphPadding(PresenterObject *self, void *closure) {
+    if (!self->_presenter) {
+        PyErr_SetString(PyExc_AssertionError, "Underlying C++ presenter is nullptr");
+        return nullptr;
+    }
+    return PyLong_FromLong(self->_presenter->graphPadding);
+}
+
+char graphPaddingName[] = "graphPadding";
+PyGetSetDef presenter_getsetters[] = {
+    {yPosName, reinterpret_cast<getter>(presenter_getYPos)},
+    {graphSizeName, reinterpret_cast<getter>(presenter_getGraphSize)},
+    {graphPaddingName, reinterpret_cast<getter>(presenter_getGraphPadding)},
+    {}};  // Sentinel
+
+char monitors_extension_doc[] = "The module is a wrapper over C++ monitors. It guarantees that C++ and Python "
+    "monitors are consistent.";
+
+PyType_Slot presenterSlots[] = {
+    {Py_tp_dealloc, reinterpret_cast<void*>(presenter_dealloc)},
+    {Py_tp_doc, monitors_extension_doc},
+    {Py_tp_methods, presenter_methods},
+    {Py_tp_getset, presenter_getsetters},
+    {Py_tp_init, reinterpret_cast<void*>(presenter_init)},
+    {Py_tp_new, reinterpret_cast<void*>(PyType_GenericNew)},
+    {}};  // Sentinel
+
+PyType_Spec presenterSpec{"monitors_extension.Presenter", sizeof(PresenterObject), 0, 0, presenterSlots};
+
+PyModuleDef monitors_extension{PyModuleDef_HEAD_INIT, "monitors_extension", monitors_extension_doc, 0};
+}
+
+PyMODINIT_FUNC PyInit_monitors_extension() {
+    import_array();
+    if (PyErr_Occurred()) return nullptr;
+
+    PyObject *presenterType = PyType_FromSpec(&presenterSpec);
+    if (!presenterType) return nullptr;
+
+    PyObject *m = PyModule_Create(&monitors_extension);
+    if (m == nullptr) return nullptr;
```

Comment:
You need to decref the type here so that it's not leaked.

Reply:
Added